Configuration is set only for plans (no longer per version and plan) (#1044)
kyma-project/kyma-environment-broker: docs/contributor/02-40-broker-configuration-for-given-plan.md

> [!NOTE] 

> > This document has been deprecated. Kyma version configuration is not possible.

Reviewer: You can delete this note now.

> Some processes in Kyma Environment Broker (KEB) can be configured to deliver different results. KEB needs a ConfigMap with a configuration for the given Kyma plan to process the requests. 

Reviewer: ```suggestion
Some Kyma Environment Broker (KEB) processes can be configured to deliver different results. KEB needs a ConfigMap with the configuration for a given Kyma plan to process the requests. 
```

> A default configuration must be defined. This configuration must be recognized by KEB as applicable for all the supported plans. You can also set a separate configuration for each plan.

Reviewer: ```suggestion
The default configuration must be defined. KEB must recognize this configuration as applicable to all supported plans. You can also set a separate configuration for each plan.
```

> While processing requests, KEB reads a configuration from a ConfigMap which holds data for a given plan.

Reviewer: ```suggestion
While processing requests, KEB reads the configuration from the ConfigMap that holds data for a given plan.
```

> [!NOTE] 

> > Create all configurations in the `kcp-system` namespace.

Reviewer: Can you explain briefly why?

Author: this note is not necessary, I'll remove it

> [!NOTE] 

> > Currently, only the Kyma custom resource template and the additional components list can be configured.

Reviewer: You have removed the `additional components` from the code, so is the info about them in this note still valid?

## ConfigMap  

The appropriate ConfigMap is selected by filtering the resources using labels. KEB recognizes the ConfigMaps with configurations when they contain the label:

```yaml
keb-config: "true"
```

> [!NOTE] 

> > Each ConfigMap that defines a configuration must have both labels assigned.

Reviewer: ```suggestion
> Each ConfigMap that defines the configuration must have this label assigned.
```

> [!NOTE] 

> > `kyma-template` and `additional-components` configurations are required.

Reviewer: ```suggestion
> The `kyma-template` and `additional-components` configurations are required.
```

> See the example of a ConfigMap with a default configuration for Kyma and specific configurations for `plan1`, `plan2`, and `trial` plans:

Reviewer: ```suggestion
See the example of a ConfigMap with the default configuration for Kyma and specific configurations for `plan1`, `plan2`, and `trial` plans:
```
